co-simulation/fed/mosaic-carma-messenger/src/main/java/org/eclipse/mosaic/fed/carmamessenger/ambassador/CarmaMessengerInstanceManager.java

```diff
@@ -17,89 +17,180 @@
 
 import java.io.IOException;
 import java.net.InetAddress;
-import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
+import org.eclipse.mosaic.interactions.traffic.VehicleUpdates;
 import org.eclipse.mosaic.lib.CommonUtil.ambassador.CommonInstanceManager;
+import org.eclipse.mosaic.lib.geo.GeoPoint;
+import org.eclipse.mosaic.lib.objects.trafficevent.MsgerTrafficEvent;
+import org.eclipse.mosaic.lib.objects.vehicle.MsgerVehicleStatus;
+import org.eclipse.mosaic.lib.objects.vehicle.VehicleData;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 
 import gov.dot.fhwa.saxton.TimeSyncMessage;
 
 public class CarmaMessengerInstanceManager extends CommonInstanceManager<CarmaMessengerInstance, CarmaMessengerRegistrationMessage>{
 
     private static final int BRIDGE_TARGET_PORT = 5500;
     private final Logger log = LoggerFactory.getLogger(this.getClass());
+    private final Map<String, CarmaMessengerRegistrationMessage> ns3RegistrationList = new HashMap<>();
+    private final Map<String, CarmaMessengerBridgeRegistrationMessage> bridgeRegistrationList = new HashMap<>();
 
     /**
      * Callback to invoked when a new CARMA Platform instance registers with the mosaic-carma ambassador for the first time
      * @param registration The new instance's registration data
      */
-    public void onNewRegistration(CarmaMessengerRegistrationMessage registration) {
+    public void onMsgerNewRegistration(Object registration) {
         super.setTargetPort(5600);
-        if (!managedInstances.containsKey(registration.getVehicleRole())) {
+        // Check registration type and get vehicle role
+        String vehicleRole = null;
+        if (registration instanceof CarmaMessengerRegistrationMessage) {
+            vehicleRole = ((CarmaMessengerRegistrationMessage) registration).getVehicleRole();
+            log.info("Receive ns3 registration message of role " + vehicleRole);
+            if(ns3RegistrationList.containsKey(vehicleRole)){
+                log.error("Duplicate ns3 registration for vehicle {}", vehicleRole);
+                return;
+            }else{
+                ns3RegistrationList.put(vehicleRole, (CarmaMessengerRegistrationMessage)registration);
+                log.debug("Added new ns3 registration for vehicle {}", vehicleRole);
+            }
+            if(bridgeRegistrationList.containsKey(vehicleRole)){
+                log.debug("Found bridge registration of vehicle {}, making a carma messenger instance now", vehicleRole);
```

Comment:
This log says it's making a new instance now, but doesn't seem to trigger anything?